lisa/runners/lisa_runner.py

```diff
@@ -116,13 +116,21 @@ def fetch_task(self) -> Optional[Callable[[], List[TestResult]]]:
                     # no environment in used, and not fit. those results cannot be run.
                     skipped_test_results = self._skip_test_results(can_run_results)
                     if skipped_test_results:
-                        return lambda: skipped_test_results
+                        return Task(
+                            self.generate_task_id(),
+                            lambda: skipped_test_results,
+                            self._log,
+                        )
         elif available_results:
             # no available environments, so mark all test results skipped.
             skipped_test_results = self._skip_test_results(available_results)
 
             self.status = ActionStatus.SUCCESS
-            return lambda: skipped_test_results
+            return Task(
```

Comment:
There can be a lightweight method like `_report_results` to make it shared. 

```python
def _report_results(self, test_results: List[TestResult])->Task[List[TestResult]]:
    return Task(self.generate_task_id(), lambda: test_results, self._log)
```

Reply:
Actually, I think this kind of small piece of lambda created some very quick tasks. It makes the worker busy in a very short time. So it affects the loop efficiency. I have no good idea to optimize them, but not increase complexity of class design so far.